service/src/main/java/bio/terra/cbas/controllers/MethodsApiController.java

```diff
@@ -177,20 +181,36 @@ public ResponseEntity<PostMethodResponse> postMethod(PostMethodRequest postMetho
       UUID methodId = UUID.randomUUID();
       UUID runSetId = UUID.randomUUID();
 
+      GithubUrlComponents githubUrlComponents =
+          MethodUtil.extractGithubDetailsFromUrl(rawMethodUrl);
+
+      String path = githubUrlComponents.path();
+      String repository = githubUrlComponents.repo();
+      String organization = githubUrlComponents.org();
+      String branchOrTagName = githubUrlComponents.branchOrTag();
+
+      GithubMethodDetails githubMethodDetails =
+          new GithubMethodDetails(repository, organization, path, isPrivate, methodId);
```

Comment:
Based on the discussion we had, we only want to store these details when method source is Github since the purpose of `github_details` table is to store information for Github methods. When the need arises for storing more details related to dockstore method, we can create a new table for it.